client/core/simnet_trade.go

```diff
@@ -1857,6 +1870,15 @@ func (s *simulationTest) assertBalanceChanges(client *simulationClient) error {
 			quoteFees = int64(fees.Swap)
 			baseFees = int64(fees.Redemption)
 		}
+		// eth based assets will grossly overestimate swap fees here.
```

Comment:
Why is this do you think?  Is it not giving fees using the current base fee, which we cannot go below?

Also, using too high fees was causing the balance failures?  Does that mean the expected balance change was computed based on different fee rates than these?

Reply:
For ethereum, the max fee rate is the only value we have when we set these fee values. This is in contrast to BTC, where the server reports the fee rate that will ultimately be used.

We **should** figure out how to update these or to populate them when we check our confirmations somehow.

Reply:
> Why is this do you think?

I think we just decided to use the max here because it was easiest. There was a pr recently that made this fail I think but cannot find it atm... we were probably using the 6 blocks thing before.

200 doesn't seem too ridiculous... https://ycharts.com/indicators/ethereum_average_gas_price Even above that at times on mainnet.

Don't all coins lock enough fees for `ord.DEXConfig.MaxFeeRate` which is the max fee from the dex config file? I'm actually not sure atm... I mean when orders are placed, before matched.

Reply:
Oh, these are the fees at match time.

Reply:
Yeah, different because https://github.com/decred/dcrdex/blob/fb05984dc0559f91f0c9d09c3fa82460e5615fae/server/dex/feemgr.go#L111-L119

Reply:
I guess the test would fail since #1447 as we were using `SuggestGasPrice` before.

Reply:
I don't guess the server's FeeRate is ever used for eth?

Reply:
I recall https://github.com/decred/dcrdex/pull/1447 and I believe I may have suggested that change because with dynamic fees it's not a problem at all.  200 is ok, and even higher is fine as MaxFeeRate (and the match's prescribed fee rate) for the same reason.

@buck54321 stated the real issue in his comment.  When the dynamic eth txns are mined, they use a different fee than what was specified when it was authored and broadcasted.  i.e. with ETH, the prescribed swap fee is not what the transaction actually consumes in fees, so if we want to assert balance change we should base it on the _real_ fees.

Therefore, the actual bug is that `core.Order.FeesPaid` does not have the correct fees paid as it should.   If we run an ETH swap and look at the completed order in the order page, it'll show it paid higher fees than actual because of this.  As per Buck's comment, we need to update `trackedTrade.metaData.{SwapFeesPaid,RedemptionFeesPaid}` once we observe the txns are mined.

To make the test "work" for now I'm ok with this PR's change, but we'd want to open an issue to do the needed update.

Reply:
Computing the fees paid on an arbitrary txns can be quite tough actually.  For UTXO assets it often means fetching the prevout values of the inputs, so for whatever solution we reach, we should make it easy on ourselves and skip the fee computation and update if it's not `SupportsDynamicTxFee` because it's not needed.  Smells like another `WalletTrait`, like `DynamicTxFeeGetter`

Reply:
look ok? #1741

Reply:
Thanks.  I started exploring the idea here https://github.com/decred/dcrdex/commit/0896ffd993856889ae42f7b7378439bcf09ab2bc
Anyone is welcome to pick it up.  I should focus one reviewing existing PRs.